lib/http.js

```diff
@@ -53,21 +53,21 @@ function httpRequest(sdk, options) {
   };
 
   var err, res;
-  return new Q(sdk.options.ajaxRequest(method, url, ajaxOptions))
+  return new Q(sdk.options.request(method, url, ajaxOptions))
     .then(function(resp) {
-      res = resp.responseText;
+      res = resp.responseText || resp;
       if (res && util.isString(res)) {
         res = JSON.parse(res);
       }
 
       if (saveAuthnState) {
         if (!res.stateToken) {
-          cookies.deleteCookie(config.STATE_TOKEN_COOKIE_NAME);
```

Comment:
Curious why `stateToken` has to be save to cookie?
With your refactor, looks like it's saved to NodeCache in Node env and wont be sent in API as cookie.
Is this intentional and will not be problematic?

Reply:
`stateToken` has to be saved to resume a transaction. It is retrieved from storage and passed as an argument here: https://github.com/okta/okta-auth-js/pull/184/files#diff-1a136b779b30ddb4520bd33f89a5c8afR48, so should work also for Node.